src/ui/components/keyboard_switch.c

```diff
@@ -82,14 +86,17 @@ static void _on_event(const event_t* event, component_t* component)
     case EVENT_TOGGLE_ALPHANUMERIC:
         switch (ks_data->mode) {
         case DIGITS:
-            ks_data->mode = LOWER_CASE;
+            ks_data->mode = SPECIAL_CHARS;
             break;
         case LOWER_CASE:
             ks_data->mode = UPPER_CASE;
             break;
         case UPPER_CASE:
             ks_data->mode = DIGITS;
             break;
+        case SPECIAL_CHARS:
+            ks_data->mode = LOWER_CASE;
+            break;
```

Comment:
Can we move this up as the 2nd entry - so that one can follow what the toggle order will be? It's confusing having to jump around to make sure that all cases are covered. Maybe let's reorder the values in the enum so that the order is preserved there as well?

Reply:
Good idea. Done, see new commit.